lib/laforge/activerecord.rb

```diff
@@ -52,46 +52,48 @@ def forge_diff(**forge_options)
     # Optionally pass `sources` to limit data entries used in the calculation to only those from the given sources
     def forge_attributes(attributes: nil, sources: nil)
       forged_attrs = {}
-      filter_loaded_data_entries(attributes: attributes, sources: sources, present: true).sort_by(&:priority_with_fallback).reverse.uniq(&:attribute_name).each do |data_entry|
+      filter_loaded_data_entries(attributes: attributes, source_ids: sources, present: true).sort_by(&:priority_with_fallback).reverse.uniq(&:attribute_name).each do |data_entry|
         forged_attrs[data_entry.attribute_name] = data_entry.value
       end
 
       return forged_attrs
     end
 
     # Record several pieces of information from the same source.
-    def record_data_entries(attributes_hash, source, **data_entry_options)
+    def record_data_entries(attributes_hash, source_name, **data_entry_options)
+      source_id = DataSource.find_by(name: source_name)&.id
```

Comment:
What happens if we don't have a source by that name? I don't think we should just keep going. Do you?

Reply:
Do we have a validation on source names to ensure they are unique?